res/controllers/Denon-MC7000-scripts.js

```diff
@@ -816,25 +849,78 @@ MC7000.sortLibrary = function(channel, control, value) {
 // VuMeters only for Channel 1-4 / Master is on Hardware
 MC7000.VuMeter = function(value, group) {
     var deckNumber = script.deckFromGroup(group),
-        vuLevelOutValue = engine.getValue(group, "PeakIndicator") ? MC7000.VuMeterLEDPeakValue : Math.pow(value, 4) * (MC7000.VuMeterLEDPeakValue - 1);
-
-    midi.sendShortMsg(0xB0 + deckNumber - 1, 0x1F, vuLevelOutValue);
+        currentLED = Math.pow(value, 4) * (MC7000.VuMeterLEDPeakValue - 1),
+        vuLevelOutValue = engine.getValue(group, "PeakIndicator") ? MC7000.VuMeterLEDPeakValue : currentLED,
+        vuThreshold = Math.floor(currentLED / 6) * 6;
+    // only send new Midi signal when playing and threshold is met
+    if (engine.getValue(group, "play")) {
+        if (MC7000.prevVuLevel[deckNumber - 1] !== vuThreshold) {
+            midi.sendShortMsg(0xB0 + deckNumber - 1, 0x1F, vuLevelOutValue);
+            MC7000.prevVuLevel[deckNumber - 1] = vuThreshold;
+        }
+    } else { // if not playing set zero, otherwise 1st LED stays active when < threashold
+        midi.sendShortMsg(0xB0 + deckNumber - 1, 0x1F, 0x00);
+    }
 };
 
-/* LEDs around Jog wheel */
-MC7000.JogLed = function(value, group) {
+// Spinning Platter LEDs & Slicer Loop PAD LEDs as beat counter
+// pulled together for the calculation to be done only once and then
+// send LED signals to Jog or PAD LEDs when needed.
+MC7000.TrackPositionLEDs = function(value, group) {
     // do nothing before track starts
     if (value === 0) {
         return;
     }
```

Comment:
Is there a rationale behind this behavior?

Reply:
YES - the value is `playposition` and I don't want the platter LEDs to move while the track is in empty space in front of the actual start of track.

Reply:
Right, but why is that a problem? IMO as with the behavior of the VuMeters, its rather unusual.
Is this the default behavior of the controller when used with other DJ software?

Reply:
Well - it is disturbing that the "needle" moves before the track actually starts. I found that in another mapping and really liked it. And tbh, I don't know and don't mind what the behavior is for other DJ Software for 2 reasons... 1st: i dont run other software as the commercial ones dont support Linux (I run Linux only for more than 15 years) and 2nd: Whatever the behavior would be in Serato or the likes then I would not copy it but instead make it better for Mixxx.

But interesting that Jan and Be never had an issue with that but now when i made a few changes the initial code is on question again. That makes it very hard to keep focus and keep me motivated to further improve it. Anyhow - I thank you for helpful hints - but several of them are just down to personal taste - I had the same discussion with Jan with my last update - some comments are relevant and some just depend on taste of the reviewer and i dont change them back and forth depending who is reviewing the changes. Isn't a rant just clarification. Thanks again.

Reply:
I agree. I'm definitely polluting this code review with my personal opinion of how a mapping should look like. I guess its best to take this discussion to Zulip so we can establish some guidelines how we would like a mapping to look like.

You are welcome to keep the code as-is. I'm just saying that its non-standard in my opinion and that should be documented in the wiki/manual page of the mapping for users that might not expect this behavior.

Reply:
That is no issue to add another sentence in the [manual](https://manual.mixxx.org/2.3/en/hardware/controllers/denon_mc7000.html#leds), where platter LEDs are descriped. I gonna make a PR for the mapping soon